app/Bus/Handlers/Traits/StoresMeta.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of Cachet.
+ *
+ * (c) Alt Three Services Limited
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CachetHQ\Cachet\Bus\Handlers\Traits;
+
+use CachetHQ\Cachet\Models\Meta;
+
+trait StoresMeta
+{
+    /**
+     * Stores all Meta values of a model.
+     *
+     * @param array      $metaData
+     * @param string     $metaType
+     * @param string|int $metaId
+     * @param string     $metaModel
+     *
+     * @return void
+     */
+    public function storeMeta($metaData, $metaType, $metaId, $metaModel = Meta::class)
+    {
+        // Validation required instead of type hinting because it could be passed as false or NULL
+        if (!is_array($metaData)) {
+            return;
+        }
+
+        foreach ($metaData as $key => $value) {
+            $modelInstance = call_user_func(
```

Comment:
This can be `$metaModel->firsOrNew(...)`.

Reply:
I could try a static call using `$metaModel::firsOrNew(...)` would that be the preferred solution?